### DISCUSSION TITLE
Resources for Beginners

### DISCUSSION THREAD

**Discussion by PixxelFixx** [joeycastillo/Sensor-Watch#51] [General]
Hi all

I've always loved the F-91W and really enjoy modding them, but have only done so in the cosmetic sense so far (@pixel.fixx on Instagram if you're interested!).

This project is really exciting and I'm looking forward to trying it out. However, I am COMPLETELY new to this type of modding i.e. coding so it's all a bit overwhelming. Would anyone be able to recommend any resources for me to brush up on the basics, or any similar kits that I can play around with? I'm keen to learn! And I mean it when I say that I am very new to this field, so resources aimed at children probably wouldn't be too far off haha. 

Thank you!

**Comment by bademux** [comment]
I think [this article](https://www.crowdsupply.com/oddly-specific-objects/sensor-watch/updates/how-we-got-to-blinky-a-tour-of-our-circuit-board) will be wonderful start. Then just google unknowns.

**Reply by PixxelFixx** [reply]
Yes this is a great starting point, very well written and thought out.

**Comment by joeycastillo** [comment]
That article is definitely a deep dive into the how it works, but for a beginner, I think the big thing is just knowing how to take apart the watch, swap out the board, and put it back together. I'm currently reworking the documentation for that (and editing a video together that shows all the steps!) but in the meantime, the [Sensor Watch Quick Start Guide](https://github.com/joeycastillo/Sensor-Watch/files/8008505/Sensor.Watch.Quick.Start.pdf) is the piece of paper that I sent out to beta testers. There's also an [earlier video](https://mobile.twitter.com/josecastillo/status/1413306113412452359) that shows a shortened version of the process.

I'd be curious to know what you think of these resources — in particular if you think that after seeing these notes, you'd feel confident disassembling and reassembling your Casio with the Sensor Watch board swap. I'm going to be rewriting this and making some new printed instructions soon, so any thoughts you have would really help. 

Thanks!

**Reply by PixxelFixx** [reply]
The quick start guide is a really good summary of functions and the diagrams are super helpful. I'm confident at taking the watch apart, as I've done it a number of times to change polarisers, solder new LEDs and insert filters etc. so I'm not too worried about that. I watched your video and I think it's easy to follow for someone who's never done it before. 

I guess my concern mainly lies with how to write specific code for a function that I would like the watch to do, and also designing flex PCBs for those particular functions. But I see there are already ones written. For example, I was particularly interested in an accelerometer function and can see there's some work already done around it. I suppose for me it's trying to understand what I'm looking at!

**Reply by joeycastillo** [reply]
ah, got it! Okay, I need to make a guide for all the resources we have, but there are three primary things I'd look at: 

* The [week three backer update](https://www.crowdsupply.com/oddly-specific-objects/sensor-watch/updates/week-three-coding-a-new-watch-face-and-another-bike-ride) is eventually going to be the new Movement README; it's the most up-to-date example of coding a new watch face. 
* The [Sensor Watch Library Documentation](https://joeycastillo.github.io/Sensor-Watch/) has a description of all of the functions in the library, which is how you interact with the hardware. Everything from the clock and the buttons to light, sound and deep sleep are covered there. 
* The [Sensor Watch Documentation](https://joeycastillo.github.io/Sensor-Watch-Documentation/) is a work in progress, but currently it's aimed at offering higher-level guidance about how you can make the most of the limited resources available on the watch, and it's where I'll eventually have more detailed guidance about building add-on boards. 

The accelerometer is going to be an area of active development soon; I sense that an LIS2DW12 accelerometer board will be my next project once I get the Crowd Supply orders squared away, as there are some fun possibilities with machine learning that I really want to explore. I'll definitely share more once I have some more work done in that direction.

**Reply by PixxelFixx** [reply]
This is great, I really appreciate you taking the time to highlight a few things for me. I'm new to GitHub too so just familiarising myself with the layout etc. 

I'll keep my eyes peeled for further updates! Thank you for all your hard work, I'm looking forward to learning a lot more out of this project!
